### ISSUE TITLE
Feature Request: Blend Modes

### ISSUE THREAD

**Issue by jeffski** [OpenShot/libopenshot#320]
I've been thinking about this for a while and might take a look at if I get a chance - blend modes.

Photoshop, Premiere and other video/photo editing software have [blend modes](https://helpx.adobe.com/au/premiere-pro/using/blending-modes.html) where you can control the way images blend over each other. With video it's really effective for overlaying light leaks, weather effects etc..

I noticed that Qt QPainter has [CompositionModes ](https://doc.qt.io/qt-5/qpainter.html#CompositionMode-enum) and several of these equate to what you would see in Adobe products (screen, multiply, dodge, etc...).

Also it seems that [Timeline ](https://github.com/OpenShot/libopenshot/blob/e4d622e7de4cc7d5c1ac1c1a553cdd89f1763f4f/src/Timeline.cpp#L606) and [Frame ](https://github.com/OpenShot/libopenshot/blob/04d1a5849a2abca7ed1255eb9e9b24d4c3eee447/src/Frame.cpp#L644) already use this:

```
painter.setCompositionMode(QPainter::CompositionMode_SourceOver);
```

So I'm wondering if this could be set by the user to a selection of sane presets to effectively add blend modes to libopenshot/OpenShot?

Any pointers or advice much appreciated and I can start investigating.

**Comment by SG57** [comment]
Any work done on this?  I may do it myself, seems silly not to support this.

**Comment by nilp0inter** [comment]
I find this feature quite essential for working with overlays. Right now the only option is to do some trickery with the alpha settings and maybe chroma effect, but it doesn't look good.

I may get this a try over the weekend. @jeffski , @SG57 have you done any work on this?

I guess we'll need a new attribute in the Clip class for settings the blend mode that would be used in `Timeline::add_layer` to set the appropriate composition mode. Any thoughts?

**Comment by tdbe** [comment]
TL;DR & PSA: Blender in the past 6 years has become a solid alternative to DaVinci Resolve, yes, for VIDEO EDITING, Blender: [https://www.youtube.com/watch?v=99ZdOuJMrao](https://www.youtube.com/watch?v=99ZdOuJMrao)

Justification why I post this:

------------


Oh wow, considering every videogame shader pipeline supports blending mode operations, and every photo and video editing program as well, and that it's been 6+ years, this is a disheartening and questionable thing to hear about the way this project was and is developed.

I've been using and learning OpenShot for a few years and providing feedback, and loved that there's an android version. But I'm sorry, between the complete lack of developer feedback on any GH issue, and these fundamental libopenshot kneecaps and the dealbreaking openshot gui's crashes when editing edited segments, and basic UX dumbness (e.g. can only speed up by 2x 4x 16x via UI, and otherwise there's random gaps in the exported video between ripple-deleted clip segments) this is just not competent enough from the ground-up, nor do they seem to be using their own product - I dare the dev & contrinuitors to cut a trailer for OpenShot's features, using OpenShot!

**Comment by Colorjet3** [comment]
Hello @tdbe 

Just so that you are aware, this is a very small project and very dependent on volunteers and not much funding.  The lead developer does his best to maintain the software (enhancements and bug fixes).  Yes, things move slowly and not always the most graceful, but it keeps moving forward.

It would be great if you can provide more details on some of the issues.  A post per issue would be great and I can put those in the queue for the lead developer to look into when time permits:

1.  GUI crashes.  Details please.  How? what causes it? 
2.  Speeb up 2x, 4x, etc.   I hear you.  This is in the queue.  And by the way, if you download v3.4.0 or the latest dev daily build, you will see a new option named "Enable Timing".  It is a clock icon on the timeline tool bar.  When you enable it, you can grab the rigth edge of a video and adjust it to the left (speed up) or right(slow motion).  It is work in prorgress as it is difficult to ascertain what speed you are achieving without manual calculation (Time attribute in the properties of the clip you are adjusting).  This is work in-progress and will further improve, in time.
3. Random gaps in the exported video between ripple-deleted clip segments.  Please provide the steps to reproduce this.  This is a critical issue that should be fixed immediately.

I am just a volunteer user and provide assitance when I can.  I also try to capture the enhancement requests and issues and report them back to the lead developer.

If you are a developer yourself, or are willing to be a tester, that would be great.  Feel free to connect with @jonoomph.

**Comment by tdbe** [comment]
>     1. GUI crashes.  Details please.  How? what causes it?

>     3. Random gaps in the exported video between ripple-deleted clip segments.  Please provide the steps to reproduce this.  This is a critical issue that should be fixed immediately.

1 is already covered in issues except for this scenario: in v3.4.0 with the clock, if you try to re-cut and/or re-time-edit segments that have been already time-edited (also try after re-opening the project)

3 also happens in v3.4.0; with segments that you time edit (and then cut) in various random ways. The minuscule gaps don't show when playing in the timeline in the gui, it only shows in the export and there's no way to fix them.

> I am just a volunteer user and provide assitance when I can. I also try to capture the enhancement requests and issues and report them back to the lead developer.
> 
> If you are a developer yourself, or are willing to be a tester, that would be great. Feel free to connect with [@jonoomph](https://github.com/jonoomph).

See, this is why I am DONE. The ONLY responses on this whole project, over a decade, come from non-devs. You have all been great enough, except it is more than abundantly clear that the programmer contributors have NEVER cut a trailer in their own software ever, in addition to not knowing how to think about developing a video editor not even theoretically (see my previous examples).

🫡
